packages/react-i18n/src/test/i18n.test.ts

```diff
@@ -346,6 +346,52 @@ describe('I18n', () => {
     });
   });
 
+  describe('#unformatCurrency()', () => {
+    const mockSymbolResult = {
+      symbol: '$',
+      prefixed: true,
+    };
+
+    it('handles formatted USD input', () => {
+      getCurrencySymbol.mockReturnValue(mockSymbolResult);
+
+      const i18n = new I18n(defaultTranslations, defaultDetails);
+      expect(i18n.unformatCurrency('1,234.50', 'USD')).toBe('1234.50');
+      expect(i18n.unformatCurrency('1', 'USD')).toBe('1.00');
+    });
+
+    describe('prefixed symbols', () => {
+      it('handles prefix with a space', () => {
+        getCurrencySymbol.mockReturnValue(mockSymbolResult);
+
+        const i18n = new I18n(defaultTranslations, defaultDetails);
+        expect(i18n.unformatCurrency('$ 1,234.50', 'USD')).toBe('1234.50');
+      });
+
+      it('handles prefix without a space', () => {
+        getCurrencySymbol.mockReturnValue(mockSymbolResult);
+
+        const i18n = new I18n(defaultTranslations, defaultDetails);
+        expect(i18n.unformatCurrency('$1,234.50', 'USD')).toBe('1234.50');
+      });
+    });
+
+    it('unformats formatted input to 2 digits', () => {
```

Comment:
Can you add tests with currencies that do not have digits (JPY) and some that have 3 digits (JOD)? 🙏